src/Provider/AbstractProvider.php

```diff
@@ -422,7 +422,8 @@ protected function appendQuery($url, $query)
         $query = trim($query, '?&');
 
         if ($query) {
-            return $url . '?' . $query;
+            $glue = strstr($url, '?') === false ? '?' : '&';
+            return $url . $glue . $query;
```

Comment:
This makes sense but raises the question as to what is supposed to happen with `$url` already contains some of the parameters in the `$query`?